src/core/namespace.c

```diff
@@ -2079,10 +2087,21 @@ int setup_namespace(
                                 .flags = DEV_MOUNT_OPTIONS,
                         };
 
+                /* In case /proc is successfully mounted with pid tree subset only (ProcSubset=pid), the
+                   protective mounts to non-pid /proc paths would fail. But the pid only option may have
+                   failed gracefully, so let's try the mounts but it's not fatal if they don't succeed. */
+                bool ignore_protect_proc = ns_info->ignore_protect_paths || ns_info->proc_subset == PROC_SUBSET_PID;
```

Comment:
This means we might get false positives and skip these on newer kernels, to support older kernels, right?

Reply:
The mounts aren't skipped, but failures to install protective mounts in `/proc` are ignored when `ProcSubset=pid` is also active. The paths like `/proc/acpi` don't exist in that case, so the mounts are useless. Of course the failures could happen due to other reasons and before this, the unit then failed but now it will succeed. I think that change isn't very dramatic.

On older kernels, `ProcSubset=pid` would fail but the failure is gracefully ignored and the protective mounts would succeed. There's no change with this commit.

On new kernels, this allows to enable both `ProcSubset=pid` together with `ProtectKernelTunables=yes` and/or `ProtectKernelLogs=yes`, which was so far only possible on older kernels. The useless protective mounts for `/proc` would not be installed, but the other mounts for `/dev` or `/sys`, and the seccomp rules are still applied, which is good.